File: java/src/test/java/regression/framework/page/objects/Homepage.java
package regression.framework.page.objects;

import org.openqa.selenium.WebDriver;
import regression.framework.SharedDriver;
import regression.framework.support.WebDriverCustomMethods;
import static regression.framework.SharedDriver.*;

public class Homepage {
    private final SharedDriver driver;

    public Homepage(WebDriver driver) {
        WebDriverCustomMethods.navigateTo(url);
        WebDriverCustomMethods.verifyPageTitle("");

        this.driver = (SharedDriver)driver;
    }
}

File: dotnet/ScribeStar.Regression.Tests/ScribeStar.Regression.Tests/Pages/Homepage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;
using ScribeStar.Regression.Tests.Core;
using ScribeStar.Regression.Tests.Core.Utils;

namespace ScribeStar.Regression.Tests.Pages
{
    public class Homepage
    {
        private SharedDriver sharedDriver;

        [FindsBy(Using = "", How = How.CssSelector), CacheLookup]
        private IWebElement logo { get; set; }

        public Homepage(IWebDriver driver)
        {
            driver.Navigate().GoToUrl(EnvironmentConfiguration.GetEnvironment());
            AssertionHandlers.BrowserTitleShouldContain("");

            this.sharedDriver = (SharedDriver)driver;
        }
    }
}
